xsdata/codegen/container.py

```diff
@@ -54,6 +55,7 @@ def __init__(self, config: GeneratorConfig):
                 FlattenAttributeGroups(self),
             ],
             Steps.FLATTEN: [
+                UnwrapLists(self),
```

Comment:
The order matters, doing this before processing types and extensions might be too early, we might need to experiment with this a bit but my first thought is that it needs to go lower much lower, perhaps after the compound fields handler